### PR TITLE
Add Appointment table and CRUD.

### PR DESCRIPTION
**Pull request by laurandarius**

### CODE REVIEW

**Comment by laurandarius** [comment]
[Add Appointment entity + CRUD.](https://trello.com/c/AVLsXnH7/16-add-appointment-entity-crud)

File: app/services/appointment.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+      error.throwValidationError('Invalid appointment format.');
+    }
+
+    const appointmentExists = await this.repository.getByPropsNonParanoid({ ...appointment });
+    if (appointmentExists) {
+      error.throwValidationError('Appointment already exists.');
+    }
+
+    if (companyId !== undefined) {
+      const fieldExists = await companyService.get(companyId);
```
Comment (lgnstn) [inline app/services/appointment.js]: This variable is not used. Please remove it.

File: app/services/appointment.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+const companyService = require('./company');
+const error = require('../services/error');
+
+class AppointmentService extends CrudService {
+  async create(appointment, companyId) {
+    if (appointment.id !== undefined) {
+      error.throwValidationError('Invalid appointment format.');
+    }
+
+    const appointmentExists = await this.repository.getByPropsNonParanoid({ ...appointment });
```
Comment (lgnstn) [inline app/services/appointment.js]: I think here you should also validate that this appointment doesn't overlap with another one.

**Review by lgnstn — CHANGES_REQUESTED** [review]

**Comment by laurandarius** [comment]
@lgnstn updated.

**Review by lgnstn — APPROVED** [review]

**Comment by lgnstn** [comment]
@laurandarius Conflicts.

**Comment by laurandarius** [comment]
@lgnstn updated.

**Pull request merged**